### DISCUSSION TITLE
Pricing - How will it work?

### DISCUSSION THREAD

**Discussion by snake-py** [vikejs/vike#2009] [Discussion] [answered]
Hey I just saw that Vike will be priced for companies. I think it makes sense to generate consistent revenue!  

I have some questions:

- how will the pricing look? Will there be a free tier? Is there already a resource, where I can read about it?
- Is there any public discussion available where one can participate or see the thoughts behind the decision? 
- How can you entforce pricing using MIT license isn't that contradictory? 
- Did you consider simply offering premium features? (e.g. premium templates, components, utilities)
- Did you consider managed deployments as an income source? (similar to Nextjs)

**Comment by brillout** [comment, marked as answer]
https://github.com/vikejs/vike/commit/3168305bd27a61d3566dad97522ed06ccf3de3e2 (npm is down so it isn't live at https://vike.dev/pricing yet).

**Reply by snake-py** [reply]
I see, so basically it is a forced sponsorship for anyone who is generating revenue with Vike ✅

**Reply by snake-py** [reply]
About the technical implementation, I see many ways how one could get around that. Even without reading the source code, simply intercepting the request would do. 

What about premium packages or features have you thought about those? 

E.g.

- AuthFlow Components
- Sever Deployment Pipelines

**Reply by brillout** [reply]
Yea, it's probably going to be a little bit of a cat-and-mouse game. Let's see how many will try to cheat.

We'd like to keep the code 100% open source (for many reasons). I believe the Open Source Pricing model is superior to the Open Core model. Let's see.
